wsjson/wsjson_js.go

```diff
@@ -0,0 +1,58 @@
+// +build js
```

Comment:
Unless you know that a file is compatible with [GopherJS](https://github.com/gopherjs/gopherjs) you usually want to use the build tag `+build js,wasm`. The addition of `wasm` clarifies for the Go compiler that this file is compatible with WebAssembly but tells the GopherJS compiler that the file is not compatible with GopherJS. This [blog post explains the situation in a little more detail](https://medium.com/gopherjs/future-of-gopherjs-and-go-in-the-browser-cc1667468160).

Basically it works like this:

|  Build tag  | Vanilla Go | WebAssembly | GopherJS |
| ---------- | ---------- | -------------- | ---------- |
| (none)       |  :white_check_mark:  | :white_check_mark: | :white_check_mark: |
| `js`            | :x:              |  :white_check_mark: |  :white_check_mark: |
| `!js`           |  :white_check_mark: | :x: | :x: |
| `js,wasm`  | :x: |  :white_check_mark: | :x: |
| `js,!wasm` | :x: |  :x: |  :white_check_mark: |

Reply:
The library should just work with GopherJS once https://github.com/gopherjs/gopherjs/issues/929 is closed.

Reply:
It won't even compile on GopherJS versions that don't support Go 1.13 because we use the new wrapping functions in the errors package.

Reply:
reason being syscall/js is shimmed by GopherJS so that it works the same as Go Wasm.

Reply:
Gotcha. Seems fine if you are not supporting versions older than 1.13.